sycl/test/on-device/sub_group_algorithms/support.h

```diff
@@ -0,0 +1,23 @@
+#include <CL/sycl.hpp>
+using namespace sycl;
+
+bool isSupportedDevice(device D) {
+  std::string PlatformName = D.get_platform().get_info<info::platform::name>();
+  if (PlatformName.find("CUDA") != std::string::npos)
```

Comment:
may be it is safer to check backend instead of string comparisons? e.g. https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Basic/get_backend.cpp

Reply:
The main check being performed here is for OpenCL 2.0 (versus OpenCL 1.0).  I could use `get_backend` for CUDA and LevelZero, but unless we have some other way of vetting the OpenCL version, we'll still end up doing string comparisons.   Do we have anything for the OpenCL version check?